### PR TITLE
Formspec: recreate item_image_button pressed state for its image and …

### PR DESCRIPTION
**Pull request by RealBadAngel**
…label

### CODE REVIEW

**Comment by kahrl** [comment]
:+1:

File: src/guiFormSpecMenu.cpp [hunk 1]
```diff
@@ -2442,6 +2433,12 @@ void GUIFormSpecMenu::drawMenu()
 		core::rect<s32> imgrect(0, 0, spec.geom.X, spec.geom.Y);
 		// Viewport rectangle on screen
 		core::rect<s32> rect = imgrect + spec.pos;
+		if (spec.parent_button && spec.parent_button->isPressed()) {
+			core::dimension2d<s32> dim = core::dimension2d<s32>(
```
Comment (nerzhul) [inline src/guiFormSpecMenu.cpp]: no assignment, please usr the constructor in dim object directly

File: src/guiFormSpecMenu.cpp [hunk 2]
```diff
@@ -2442,6 +2433,12 @@ void GUIFormSpecMenu::drawMenu()
...
+				skin->getSize(gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_X),
+				skin->getSize(gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_Y));
+			rect += dim;
```
Comment (nerzhul) [inline src/guiFormSpecMenu.cpp]: and do directly the += operation on the core::dimensiond2, don't assign a useless variable

File: src/guiFormSpecMenu.cpp [hunk 3]
```diff
@@ -2474,8 +2471,17 @@ void GUIFormSpecMenu::drawMenu()
...
 	for (u32 i = 0; i < m_static_texts.size(); i++) {
 		const StaticTextSpec &spec = m_static_texts[i];	
+		core::rect<s32> rect = spec.rect;
+		if (spec.parent_button && spec.parent_button->isPressed()) {
+			// Use image offset instead of text's because its a bit smaller
+			// and fits better, also TEXT_OFFSET_X is always 0
+			core::dimension2d<s32> dim = core::dimension2d<s32>(
+				skin->getSize(gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_X),
+				skin->getSize(gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_Y));
+			rect += dim;
```
Comment (nerzhul) [inline src/guiFormSpecMenu.cpp]: same here

File: src/guiFormSpecMenu.h [hunk 4]
```diff
@@ -141,24 +141,39 @@ class GUIFormSpecMenu : public GUIModalMenu
...
 			scale = true;
+			parent_button = NULL;
+		}
+		ImageDrawSpec(const std::string &a_name,
+				const v2s32 &a_pos, const v2s32 &a_geom):
+			name(a_name),
+			pos(a_pos),
+			geom(a_geom)
+		{
+			scale = true;
```
Comment (est31) [inline src/guiFormSpecMenu.h]: use the initialisator list

File: src/guiFormSpecMenu.h [hunk 5]
```diff
@@ -141,24 +141,39 @@ class GUIFormSpecMenu : public GUIModalMenu
...
 		ImageDrawSpec(const std::string &a_name,
+				const std::string &a_item_name,
 				const v2s32 &a_pos, const v2s32 &a_geom):
 			name(a_name),
+			item_name(a_item_name),
 			pos(a_pos),
 			geom(a_geom)
 		{
 			scale = true;
+			parent_button = NULL;
```
Comment (est31) [inline src/guiFormSpecMenu.h]: use the initialisator list

File: src/guiFormSpecMenu.h [hunk 6]
```diff
@@ -166,9 +181,11 @@ class GUIFormSpecMenu : public GUIModalMenu
 			pos(a_pos)
 		{
 			scale = false;
+			parent_button = NULL;
```
Comment (est31) [inline src/guiFormSpecMenu.h]: use the initialisator list

File: src/guiFormSpecMenu.h [hunk 7]
```diff
@@ -231,15 +248,26 @@ class GUIFormSpecMenu : public GUIModalMenu
...
 		StaticTextSpec()
 		{
+			parent_button = NULL;
 		}
 		StaticTextSpec(const std::wstring &a_text,
 				const core::rect<s32> &a_rect):
 			text(a_text),
 			rect(a_rect)
 		{
+			parent_button = NULL;
```
Comment (est31) [inline src/guiFormSpecMenu.h]: use the initializer list.

**Comment by est31** [comment]
5ca48a35a610aefd5c34d615cb49900f423b7aa6

**Pull request closed without merging**